libs/core/datastructures/include/hpx/datastructures/detail/small_vector.hpp

```diff
@@ -525,9 +528,18 @@ namespace hpx::detail {
         template <typename It>
         void assign(It first, It last, std::forward_iterator_tag /*unused*/)
         {
+            auto s = std::distance(first, last);
+            if constexpr (emulate_inplace_vector)
+            {
+                // Can not have an inplace_vector with a size larger than N
+                if (s <= capacity())
```

Comment:
Nit: `Cannot`
Major: Isn't this condition backwards? You mean `if (s > capacity())`.